pkg/platform/k8s/pod_controller.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// PodReconciler contain information on the clusters pods.
+type PodReconciler struct {
+	client.Client
+	ipToPod map[string]metav1.ObjectMeta
```

Comment:
`ObjectMeta` probably contains more than you need (e.g., GKV, annotations, ....) - consider creating a local struct with only the fields we need (name, namespace, labels? maybe also ServiceAccount name)

Reply:
Instead of `string` representation of the IP consider using `net/netip.Addr`?

Reply:
Agree on the first comment.
About the second one I am not sure, the values for IP in the coreV1.Pod is also string and this is the one I am using and comparing to it.